File: informatika ismeretek/kozep/2019_may/java/Versenyzo.java
public class Versenyzo {
    
    public final String nev;
    public final int rajtszam;
    public final String kategoria;
    public final String ido;
    public final int befejezesSzazalek;
    
    public Versenyzo(String line) {
        var split = line.split(";");
        
        nev = split[0];
        rajtszam = Integer.parseInt(split[1]);
        kategoria = split[2];
        ido = split[3];
        befejezesSzazalek = Integer.parseInt(split[4]);
    }
    
    public double idoOraban() {
        var daraboltIdo = ido.split(":");
        var ora = Double.parseDouble(daraboltIdo[0]);
        var perc = Double.parseDouble(daraboltIdo[1]);
        var mp = Double.parseDouble(daraboltIdo[2]);
        
        return ora + (perc / 60D) + (mp / 3600D);
    }
}

File: informatika_ismeretek/kozep/2019_may/c#/Versenyzo.cs
public class Versenyzo {

    public readonly string nev;
    public readonly int rajtszam;
    public readonly string kategoria;
    public readonly string ido;
    public readonly int befejezesSzazalek;
    
    public Versenyzo(string line) {
        var split = line.Split(';');
        
        nev = split[0];
        rajtszam = int.Parse(split[1]);
        kategoria = split[2];
        ido = split[3];
        befejezesSzazalek = int.Parse(split[4]);
    }
    
    public double idoOraban() {
        var daraboltIdo = ido.Split(':');
        var ora = double.Parse(daraboltIdo[0]);
        var perc = double.Parse(daraboltIdo[1]);
        var mp = double.Parse(daraboltIdo[2]);
        
        return ora + (perc / 60D) + (mp / 3600D);
    }
}